### PR TITLE
Create a system to minimize id conflicts

### PR DESCRIPTION
**Pull request by GiantLuigi4**
Sample usage:
```java
        Properties properties = new Properties();
        ConfigHandler[] handler = new ConfigHandler[1];
        File config = new File(FabricLoader.getInstance().getConfigDir() + "/config/bta-halplibe.properties");
        RegistryHelper.scheduleRegistry(config.exists(), ()->{
            int minBlock = BlockHelper.findOpenIds(10);
            int minItem = ItemHelper.findOpenIds(10);
            
            for (int i = 0; i < 10; i++) {
                Block bk = new BlockBuilder("halplibe").noItemBlock().build(new Block("block" + (i + 1), i + minBlock, Material.stone));
                Item item = ItemHelper.createItem("halplibe", new HalplibeItemBlock(i + minItem, bk), "block_item" + (i + 1));
            }
            
            properties.put("item_start", "" + minItem);
            properties.put("block_start", "" + minBlock);
            
            handler[0].writeDefaultConfig();
        });
        handler[0] = new ConfigHandler("bta-halplibe", properties);
```

Basically, what this PR allows for is:
A) a unified way to register items and blocks
B) a way for mods to automatically get free ids without the player editing configs (should also work fine with mods being added, assuming that said mod isn't being *re*added)
C) block ids to be desynced from item ids

### CODE REVIEW

File: src/main/java/turniplabs/halplibe/helper/RegistryHelper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package turniplabs.halplibe.helper;
+
+import java.util.ArrayList;
+
+public class RegistryHelper {
+	private static final ArrayList<Runnable> regsiterFunctions = new ArrayList<>();
```
Comment (MartinSVK12) [inline src/main/java/turniplabs/halplibe/helper/RegistryHelper.java]: There is a typo here lol

**Review by MartinSVK12 — CHANGES_REQUESTED** [review]
Besides the one typo i found, looks good? I will still test this later on my machine before merging though

File: src/main/java/turniplabs/halplibe/helper/RegistryHelper.java (hunk 1 above)
Comment (GiantLuigi4) [reply src/main/java/turniplabs/halplibe/helper/RegistryHelper.java]: If you’re referring to "regsiterFunctions", I have a tendency to put variables that people aren’t supposed to access into different forms of the word I want to use

**Review by MartinSVK12 — APPROVED** [review]

**Review by MartinSVK12 — CHANGES_REQUESTED** [review]
as discusses on the bta discord, there are additional changes that need to be made

**Comment by GiantLuigi4** [comment]
Updated usage:
```java
        Properties properties = new Properties();
        ConfigHandler[] handler = new ConfigHandler[1];
        File config = new File(FabricLoader.getInstance().getConfigDir() + "/config/bta-halplibe.properties");
        RegistryHelper.scheduleRegistry(config.exists(), ()->{
            int minBlock = BlockHelper.findOpenIds(10);
            int minItem = ItemHelper.findOpenIds(10);

            for (int i = 0; i < 10; i++) {
                Block bk = new BlockBuilder("halplibe").build(new Block("block" + (i + 1), i + minBlock, Material.stone));
                Item itm = ItemHelper.createItem("halplibe", new Item(minItem + i), "item" + (i + 1));
            }

            properties.put("item_start", "" + minItem);
            properties.put("block_start", "" + minBlock);

            handler[0].writeDefaultConfig();
        });
        handler[0] = new ConfigHandler("bta-halplibe", properties);
```

**Review by MartinSVK12 — APPROVED** [review]
lgtm

**Pull request merged**